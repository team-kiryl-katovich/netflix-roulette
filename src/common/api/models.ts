export type Result<T> = {
  data?: T;
  error?: Error;
};

export type Offset = {
  offset?: number;
  limit?: number;
};

export type OffsetResult<T> = Result<T> &
  Offset & {
    total?: number;
  };

export type Movie = {
  id: number;
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
};

export type SortBy = keyof Movie;
export type SortOrder = 'desc' | 'asc';
export type SearchBy = Extract<keyof Movie, 'title' | 'genres'>;

export type MovieFilter = {
  sortBy?: SortBy;
  sortOrder?: SortOrder;
  search?: string;
  searchBy?: SearchBy;
  filter?: string;
};
