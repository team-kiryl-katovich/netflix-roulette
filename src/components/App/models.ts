import { Store } from 'redux';

import { AppState } from '../../store/models';

export type AppProps = {
  store: Store<AppState>;
};
