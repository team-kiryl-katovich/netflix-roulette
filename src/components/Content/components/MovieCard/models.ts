export type MovieCardProps = {
  id: number;
  title: string;
  gengre: string;
  year: number;
  imageUrl: string;
};
