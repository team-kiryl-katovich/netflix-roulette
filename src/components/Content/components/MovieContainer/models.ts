import { WithMoviesProps } from 'src/enhancers/withMovies';

export type MovieContainerComponentProps = WithMoviesProps;
