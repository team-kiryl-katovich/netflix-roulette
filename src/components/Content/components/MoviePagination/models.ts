import { Movie, Offset } from '@common/api';

export type MoviePaginationProps = {
  moviesCount: number;
  offset: Offset;
  pageSize: number;
  loading: boolean;
  children?: React.ReactNode;
  fetch: (offset: Offset) => void;
};

export type MoviePaginationState = {};
