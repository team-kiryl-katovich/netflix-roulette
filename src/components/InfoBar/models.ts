import { WithFilterProps } from '@enhancers/withFilter';
import { WithMoviesProps } from '@enhancers/withMovies';

export type InfoBarComponentProps = WithFilterProps & WithMoviesProps;
