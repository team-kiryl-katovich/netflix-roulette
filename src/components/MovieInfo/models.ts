import { WithMovieProps } from '@enhancers/withMovie';

export type MovieInfoComponentProps = WithMovieProps;
