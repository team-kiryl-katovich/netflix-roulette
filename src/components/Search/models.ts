import { WithMoviesHandlersProps } from '@enhancers/withMovies';
import { WithFilterProps } from '@enhancers/withFilter';

export type SearchComponentProps = WithFilterProps & WithMoviesHandlersProps;
