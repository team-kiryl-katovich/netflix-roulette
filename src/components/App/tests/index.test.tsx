import * as React from 'react';
import * as enzyme from 'enzyme';

import { createStore } from 'redux';
import { createMemoryHistory } from 'history';

import { AppProps } from '../models';
import { App } from '..';

describe('[COMPONENTS] [APP]', () => {
  test('should render component', () => {
    const fakeProps: AppProps = {
      store: createStore(jest.fn()),
      history: createMemoryHistory({ keyLength: 0 }),
    };

    const component = enzyme.shallow(<App {...fakeProps} />);
    expect(component).toMatchSnapshot();
  });
});
