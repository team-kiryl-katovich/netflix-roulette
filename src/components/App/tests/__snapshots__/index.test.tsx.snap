// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] [APP] should render component 1`] = `
<div
  className="netflix-roulette"
>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(ConnectedRouterWithContext)
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ErrorBoundary>
        <Component>
          <Switch>
            <Route
              exact={true}
              path="/"
            >
              <Connect(Connect(Component)) />
            </Route>
            <Route
              path="/search/:query"
            >
              <Connect(Connect(Component)) />
            </Route>
            <Route
              path="/movie/:id"
            >
              <Connect(Component) />
            </Route>
            <Route
              path="/404"
            >
              <Component />
            </Route>
            <Redirect
              from="*"
              to="/404"
            />
          </Switch>
        </Component>
        <Connect(Connect(Component)) />
        <Component />
        <Component />
      </ErrorBoundary>
    </Connect(ConnectedRouterWithContext)>
  </Provider>
</div>
`;
